core/include/grids/populator.hpp

```diff
@@ -21,20 +28,29 @@ namespace detray {
  * @note bare_value and store_value are identicial in this case
  **/
 template <typename value_type = dindex,
-          value_type kInvalid = std::numeric_limits<dindex>::max(),
           template <typename> class vector_type = dvector>
 struct replace_populator {
+    DETRAY_HOST_DEVICE
+    replace_populator(const value_type invalid = invalid_value<value_type>())
+        : kInvalid(invalid) {}
+
+    value_type kInvalid;
```

Comment:
To follow convention (at least based on my observation :)), this would be better named as ```invalid_value```, since it is now a regular member